js/src/EclipseBlogs.js

```diff
@@ -0,0 +1,156 @@
+/*!
+ * Copyright (c) 2021 Eclipse Foundation, Inc.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * Contributors:
+ *  Martin Lowe <martin.lowe@eclipse-foundation.org>
+ *   Christopher Guindon <chris.guindon@eclipse-foundation.org>
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import jQuery from 'jquery';
+import template from './templates/blog-item.mustache';
+import Mustache from 'mustache';
+import dateFormat from 'dateformat';
+import 'core-js/modules/es.string.pad-start';
+
+const eclipseFdnRenderRSS = (function ($, document) {
+  function fetchRSSFeed(name) {
+    return new Promise(function (resolve, reject) {
+      $.ajax({
+        type: 'GET',
+        dataType: 'xml',
+        async: true,
+        url: name,
+      })
+        .done(function (data) {
+          let items = data.querySelectorAll('item');
+          if (items.length === 0) {
+            items = data.querySelectorAll('entry');
+          }
+
+          const blogPosts = [];
+          items.forEach((el) => {
+            const item = {
+              title: '',
+              link: '',
+              summary: '',
+              date: '',
+              id: '',
+              author: '',
+              formatDate: function () {
```

Comment:
Why do you have 2 nested functions here?

Reply:
See lambda section https://mustache.github.io/mustache.5.html